### DISCUSSION TITLE
Many isolated traces with one or two spans in Akka-Streams application

### DISCUSSION THREAD

**Discussion by bondarenko** [open-telemetry/opentelemetry-java-instrumentation#2889] [Q&A] [answered]
After adding Opentelemetry agent to my apps for automatic instrumentation I see mainly traces consisting of one or two spans. Some of them could be joined logically into one. The apps are written with Akka-streams as the main library.
Actually, I've looked into an example, [otel-demo](https://github.com/williewheeler/otel-demo), written on Spring and there I can see a span for every method call and it's automatic instrumentation as well.

![image](https://user-images.githubusercontent.com/825753/116547285-5cac8580-a8fb-11eb-913e-dc0208525679.png)

In my apps traces look as follows:

![image](https://user-images.githubusercontent.com/825753/116547655-d8a6cd80-a8fb-11eb-8551-638369fcf6f1.png)

As you can see, the traces have only one span.
Possibly that's explained by that Akka-streams isn't supported for automatic instrumentation and the opposite for Spring. What could be an explanation for such situation?

**Comment by iNikem** [comment, marked as answer]
You are absolutely correct. This project already has support for Spring and thus you have correct and nice traces from Spring application. But we don't support akka-streams at the moment thus we are unable to correctly collect and compose traces from it.
